File: app/internal/publisher/template/java/WelcomeDialog.java
package com.websmith.sdk;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import java.io.File;
import java.io.FileReader;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class WelcomeDialog {

    private final ApiClient client;
    private final HardwareDetector hardware;
    private final CacheManager cache;
    private final String productName;
    private final boolean trialEnabled;
    private final String supportEmail;

    public WelcomeDialog(ApiClient client, String productName, boolean trialEnabled) {
        this.client = client;
        this.hardware = new HardwareDetector();
        this.cache = client != null ? client.getCache() : new CacheManager();
        this.productName = productName != null ? productName : "";
        this.trialEnabled = trialEnabled;
        this.supportEmail = "support@websmithdigital.com";
    }

    public WelcomeDialog(ApiClient client, String productName) {
        this(client, productName, true);
    }

    public WelcomeDialog(ApiClient client) {
        this(client, loadProductName(client), loadTrialEnabled(client));
    }

    private static String loadProductName(ApiClient client) {
        if (client == null) return "";
        try {
            JsonObject config = loadConfigFile("config/api-config.json");
            if (config.has("product")) {
                JsonObject product = config.getAsJsonObject("product");
                if (product.has("name")) {
                    return product.get("name").getAsString();
                }
            }
        } catch (Exception ignored) {}
        return "";
    }

    private static boolean loadTrialEnabled(ApiClient client) {
        if (client == null) return false;
        try {
            JsonObject config = loadConfigFile("config/api-config.json");
            if (config.has("trial")) {
                JsonObject trial = config.getAsJsonObject("trial");
                if (trial.has("enabled")) {
                    return trial.get("enabled").getAsBoolean();
                }
            }
        } catch (Exception ignored) {}
        return false;
    }

    private static JsonObject loadConfigFile(String path) {
        try {
            File f = new File(path);
            if (f.exists()) {
                try (FileReader reader = new FileReader(f)) {
                    return JsonParser.parseReader(reader).getAsJsonObject();
                }
            }
        } catch (Exception ignored) {}
        return new JsonObject();
    }

    public boolean isOnboardingComplete() {
        return cache.isOnboardingComplete();
    }

    public Map<String, Object> show() {
        Map<String, Object> result = new HashMap<>();
        if (!trialEnabled) {
            System.out.println("Trial onboarding is not enabled for this product.");
            result.put("skipped", true);
            result.put("message", "Trial not enabled");
            return result;
        }
        if (isOnboardingComplete()) {
            System.out.println("Onboarding has already been completed.");
            result.put("skipped", true);
            result.put("message", "Already completed");
            return result;
        }

        System.out.println("\n=== Welcome to " + (productName.isEmpty() ? "Software" : productName) + " ===\n");
        System.out.println("Complete your registration to start the trial.\n");

        Scanner scanner = new Scanner(System.in);
        System.out.print("Name *: ");
        String name = scanner.nextLine().trim();
        while (name.isEmpty()) {
            System.out.print("Name is required. Please enter your name: ");
            name = scanner.nextLine().trim();
        }

        System.out.print("Email *: ");
        String email = scanner.nextLine().trim();
        while (email.isEmpty() || !email.contains("@")) {
            System.out.print("Valid email is required. Please enter your email: ");
            email = scanner.nextLine().trim();
        }

        System.out.print("Company (optional): ");
        String company = scanner.nextLine().trim();

        System.out.println("\nStarting trial for: " + name + " <" + email + ">...\n");

        try {
            String hardwareId = hardware.getFingerprint();
            Map<String, Object> customerData = new HashMap<>();
            customerData.put("company_name", company);
            customerData.put("hardware_id", hardwareId);

            com.google.gson.JsonObject extraData = new com.google.gson.JsonObject();
            extraData.addProperty("company_name", company);
            extraData.addProperty("hardware_id", hardwareId);

            com.google.gson.JsonObject trialResult = client.startTrial(email, name, hardwareId, extraData);

            if (trialResult.has("success") && trialResult.get("success").getAsBoolean()) {
                cache.setOnboardingComplete();
                System.out.println("\nTrial activated successfully! You can now use the software.\n");
                result.put("name", name);
                result.put("email", email);
                result.put("hardware_id", hardwareId);
                result.put("onboarding_complete", true);
                return result;
            }

            String errorMsg = trialResult.has("error") ? trialResult.get("error").getAsString()
                : trialResult.has("message") ? trialResult.get("message").getAsString()
                : "Unknown error";
            System.out.println("\nFailed to start trial: " + errorMsg);
            result.put("skipped", true);
            result.put("message", errorMsg);
            return result;
        } catch (Exception e) {
            System.out.println("\nError: " + e.getMessage());
            result.put("skipped", true);
            result.put("message", e.getMessage());
            return result;
        }
    }
}

File: app/internal/publisher/template/dotnet/WelcomeDialog.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;
using System.Threading.Tasks;

namespace WebsmithSDK;

public class WelcomeDialog
{
    private readonly ApiClient _client;
    private readonly HardwareDetector _hardware;
    private readonly CacheManager _cache;
    private readonly string _productName;
    private readonly bool _trialEnabled;
    private readonly string _supportEmail;

    public class WelcomeDialogResult
    {
        public bool Skipped { get; set; }
        public string? Message { get; set; }
        public string? Name { get; set; }
        public string? Email { get; set; }
        public string? HardwareId { get; set; }
        public bool OnboardingComplete { get; set; }
    }

    public WelcomeDialog(ApiClient client, string? productName = null, bool? trialEnabled = null)
    {
        _client = client;
        _hardware = new HardwareDetector();
        _cache = client?.GetCache() ?? new CacheManager();
        _productName = productName ?? "";
        bool trialEnabledVal = trialEnabled ?? LoadTrialEnabled();
        _trialEnabled = trialEnabledVal;
        _supportEmail = LoadSupportEmail();
    }

    public WelcomeDialog(ApiClient client) : this(client, null, null) { }

    private static bool LoadTrialEnabled()
    {
        try
        {
            var path = "config/api-config.json";
            if (File.Exists(path))
            {
                var doc = JsonDocument.Parse(File.ReadAllText(path));
                if (doc.RootElement.TryGetProperty("trial", out var trial)
                    && trial.TryGetProperty("enabled", out var enabled))
                {
                    return enabled.GetBoolean();
                }
            }
        }
        catch { }
        return false;
    }

    private static string LoadSupportEmail()
    {
        try
        {
            var path = "config/api-config.json";
            if (File.Exists(path))
            {
                var doc = JsonDocument.Parse(File.ReadAllText(path));
                if (doc.RootElement.TryGetProperty("branding", out var branding)
                    && branding.TryGetProperty("support_email", out var email))
                {
                    return email.GetString() ?? "support@websmithdigital.com";
                }
            }
        }
        catch { }
        return "support@websmithdigital.com";
    }

    public bool IsOnboardingComplete() => _cache.IsOnboardingComplete();

    public WelcomeDialogResult Show()
    {
        var result = new WelcomeDialogResult();

        if (!_trialEnabled)
        {
            Console.WriteLine("Trial onboarding is not enabled for this product.");
            result.Skipped = true;
            result.Message = "Trial not enabled";
            return result;
        }

        if (IsOnboardingComplete())
        {
            Console.WriteLine("Onboarding has already been completed.");
            result.Skipped = true;
            result.Message = "Already completed";
            return result;
        }

        Console.WriteLine($"\n=== Welcome to {(string.IsNullOrEmpty(_productName) ? "Software" : _productName)} ===\n");
        Console.WriteLine("Complete your registration to start the trial.\n");

        Console.Write("Name *: ");
        var name = Console.ReadLine()?.Trim() ?? "";
        while (string.IsNullOrEmpty(name))
        {
            Console.Write("Name is required. Please enter your name: ");
            name = Console.ReadLine()?.Trim() ?? "";
        }

        Console.Write("Email *: ");
        var email = Console.ReadLine()?.Trim() ?? "";
        while (string.IsNullOrEmpty(email) || !email.Contains('@'))
        {
            Console.Write("Valid email is required. Please enter your email: ");
            email = Console.ReadLine()?.Trim() ?? "";
        }

        Console.Write("Company (optional): ");
        var company = Console.ReadLine()?.Trim() ?? "";

        Console.WriteLine($"\nStarting trial for: {name} <{email}>...\n");

        try
        {
            var hardwareId = _hardware.GetFingerprint();
            var customerData = new Dictionary<string, object>
            {
                ["company_name"] = company,
                ["hardware_id"] = hardwareId
            };

            var trialResult = _client.StartTrial(email, name, hardwareId, customerData).Result;

            if (trialResult.RootElement.TryGetProperty("success", out var s) && s.GetBoolean())
            {
                _cache.SetOnboardingComplete();
                Console.WriteLine("\nTrial activated successfully! You can now use the software.\n");
                result.Name = name;
                result.Email = email;
                result.HardwareId = hardwareId;
                result.OnboardingComplete = true;
                return result;
            }

            var errorMsg = trialResult.RootElement.TryGetProperty("error", out var err)
                ? err.GetString()
                : trialResult.RootElement.TryGetProperty("message", out var msg)
                    ? msg.GetString() : "Unknown error";
            Console.WriteLine($"\nFailed to start trial: {errorMsg}");
            result.Skipped = true;
            result.Message = errorMsg;
            return result;
        }
        catch (Exception e)
        {
            Console.WriteLine($"\nError: {e.Message}");
            result.Skipped = true;
            result.Message = e.Message;
            return result;
        }
    }
}
